zjit/src/codegen.rs

```diff
@@ -1042,6 +1042,28 @@ fn gen_anytostring(asm: &mut Assembler, val: lir::Opnd, str: lir::Opnd, state: &
     Some(asm_ccall!(asm, rb_obj_as_string_result, str, val))
 }
 
+fn gen_to_array(asm: &mut Assembler, val: lir::Opnd, state: &FrameState) -> Option<lir::Opnd> {
+    // Save PC
+    gen_save_pc(asm, state);
```

Comment:
This does a lot more than just allocate -- it can run arbitrary user code. This means we have to save more state to the EC than just the PC, I think

Reply:
We may need to write out a full frame :/ cc @k0kubun and @XrXr ?